example/cajita_tutorial/10_spline/spline_example.cpp

```diff
@@ -0,0 +1,229 @@
+
+    // Get the local grid.
+    int halo_cell_width = 1;
+    auto local_grid = Cajita::createLocalGrid( global_grid, halo_cell_width );
+
+    // Get the local mesh for this rank.
+    auto local_mesh = Cajita::createLocalMesh<ExecutionSpace>( *local_grid );
+
+    // Create a particle position vector.
+    double xp[3] = { 0.5, 2, 1.5 };
+
+    /********************************************************************/
+    /*
+      There are two ways to use Cajita splines in practice.
+
+      In the first, one constructs an object of type:
+        SplineData<Scalar, Order, NumSpaceDim, EntityType, Tags...> where
+            Scalar is the underlying scalar type
+            Order = spline order
+            NumSpaceDim = number of spatial dimensions
+            EntityType = geometric entity type over which the spline is defined
+            Tags = variadic list of desired SplineDataMember tags.
+
+      The available SplineDataMembers and their associated member names are
+        SplinePhysicalCellSize = dx[d]
+        SplineLogicalPosition = x[d]
+        SplinePhysicalDistance = d[d]
+        SplineWeightValues = w[d][num_knots]
+        SplineWeightPhysicalGradients g[d][num_knots]
+
+      This design allows for customizing the relavant spline data for a
+      particular application. If all Tags are omitted, all SplineDataMembers are
+      included.
+    */
+    using value_type = double;
+    constexpr int order = 2;     // quadratic B-spline
+    constexpr int num_space = 3; // 3-dimensional
+
+    // This spline data type will be defined on grid nodes in this example.
+    using SD = Cajita::SplineData<value_type, order, num_space, Cajita::Node>;
+
+    SD sd_n; // Default construction
+    std::cout << "Spline order: " << SD::order << " (quadratic)" << std::endl;
+
+    // Evaluate the spline data given a point position and the local mesh
+    Cajita::evaluateSpline( local_mesh, xp, sd_n );
+
+    // The spline data by default contains all available SplineDataMembers,
+    // whose values were all updated from the previous call to evaluateSpline()
+    std::cout << "Spline weights:" << std::endl;
+    for ( int d = 0; d < num_space; ++d )
+    {
+        std::cout << "  Dim " << d << ": ";
+        for ( int k = 0; k < SD::num_knot; ++k )
+        {
+            std::cout << sd_n.w[d][k] << " ";
+        }
+        std::cout << std::endl;
+    }
+    std::cout << std::endl;
+
+    /*
+      We can also use the Spline<Order> interface directly to evaluate
+      specific spline data given basic information about the local grid.
+    */
+    double rdx = 1.0 / cell_size;
+    double values[3]; // num_knots for a 2nd-order spline
+    double x0[3];     // Logical position in primal grid
+
+    std::cout << "Spline weights (static interface):" << std::endl;
+    for ( int d = 0; d < num_space; ++d )
+    {
+        // These interfaces are for a single dimension only.
+        x0[d] = Cajita::Spline<2>::mapToLogicalGrid( xp[d], rdx,
```

Comment:
Maybe add a general comment that you're about to go through all the spline interfaces in order

Reply:
Done.